src/python/pants/rules/core/distdir.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from pathlib import Path
+
+from pants.base.build_root import BuildRoot
+from pants.engine.rules import rule
+from pants.fs.fs import is_child_of
+from pants.option.options_bootstrapper import OptionsBootstrapper
+
+
+class InvalidDistDir(Exception):
+  pass
+
+
+@dataclass(frozen=True)
+class DistDir:
+  """The directory to which we write distributable files."""
+  relpath: Path
+
+
+@rule
+async def get_distdir(options_bootstrapper: OptionsBootstrapper, buildroot: BuildRoot) -> DistDir:
+  global_options = options_bootstrapper.bootstrap_options.for_global_scope()
+  return validate_distdir(Path(global_options.pants_distdir), buildroot.pathlib_path)
+
+
+def validate_distdir(distdir: Path, buildroot: Path):
```

Comment:
I don't think there's any reason to omit the `-> DistDir` return type annotation here

Reply:
Ooops, that was an oversight.